binfmt/binfmt_exec.c

```diff
@@ -228,7 +230,7 @@ int exec(FAR const char *filename, FAR char * const *argv,
 {
   int ret;
 
-  ret = exec_spawn(filename, argv, exports, nexports, NULL);
+  ret = exec_spawn(filename, argv, NULL, exports, nexports, NULL);
```

Comment:
The function exec() is called directly from apps/xxx, making it incredibly difficult to obtain the environment strings.

I think the standard way is to use a global char **environ; but I'm not sure how that works under the hood. Maybe its value is updated for each context switch, or it has some base / global value.

Reply:
How about add envp to exec and modify apps/ to forward environ instead?

Reply:
Yes this I will do, now that I know how to access char **environ;